src/System.Buffers.Primitives/System/Buffers/ReadOnlyBufferReader.cs

```diff
@@ -6,21 +6,27 @@
 
 namespace System.Buffers
 {
-    public class BufferReader
+    public static class BufferReader
     {
         public static BufferReader<TSequence> Create<TSequence>(TSequence buffer) where TSequence : ISequence<ReadOnlyMemory<byte>>
         {
             return new BufferReader<TSequence>(buffer);
         }
+
+        public static int CopyTo<TSequence>(BufferReader<TSequence> reader, Span<byte> destination)
```

Comment:
Why do you need it here and not on the instance?

Reply:
It's a limitation of the C# compiler. The Span can be stack allocated but the instance CopyTo method could store the passed in span into the _currentSegment field.